packages/unmock-core/src/generator.ts

```diff
@@ -87,13 +108,27 @@ const getStateForOperation = (
   const possibleResponseCodes = Object.keys(state).filter((code: string) =>
     operationCodes.includes(code),
   );
-  // if only one response code is set - use that one; otherwise draw at random
-  // TODO - do we want to set sensible defaults?
-  const statusCode = firstOrRandomOrUndefined(possibleResponseCodes);
-  if (statusCode === undefined) {
+  if (possibleResponseCodes.length === 0) {
     return undefined;
   }
 
+  const statusCode = chooseResponseCode(
```

Comment:
Could also do `const statusCode = isFlaky ? ... : ...` to avoid the boolean argument.